utils/parse_test.go

```diff
@@ -1,35 +1,55 @@
 package utils
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
+func M(args []string, err error) []interface{} {
+	return []interface{}{args, err}
+}
+
+type ParseShellOutput struct {
+	Args []string
+	Err  error
+}
+
 func TestParseShell(t *testing.T) {
-	t.Skip("TODO: Fix")
+	cases := map[string]ParseShellOutput{
+		"":                         ParseShellOutput{[]string{}, nil},
+		"one":                      ParseShellOutput{[]string{"one"}, nil},
+		"one two":                  ParseShellOutput{[]string{"one", "two"}, nil},
+		"\"one\"":                  ParseShellOutput{[]string{"one"}, nil},
+		"'one'":                    ParseShellOutput{[]string{"one"}, nil},
+		"one two three":            ParseShellOutput{[]string{"one", "two", "three"}, nil},
+		"one \"two\" three":        ParseShellOutput{[]string{"one", "two", "three"}, nil},
+		"one \"two three\"":        ParseShellOutput{[]string{"one", "two three"}, nil},
+		"one 'two three'":          ParseShellOutput{[]string{"one", "two three"}, nil},
+		"\"one two three\"":        ParseShellOutput{[]string{"one two three"}, nil},
+		"'one two three'":          ParseShellOutput{[]string{"one two three"}, nil},
+		"‘one two three’":          ParseShellOutput{[]string{"one two three"}, nil},
+		"“one two three”":          ParseShellOutput{[]string{"one two three"}, nil},
+		"one 'two three":           ParseShellOutput{[]string{"one 'two", "three"}, nil},
+		"one \"let's have lunch\"": ParseShellOutput{[]string{"one", "let's have lunch"}, nil},
+	}
+
+	for input, expected := range cases {
+		t.Run(input, func(t *testing.T) {
+			assert.Equal(t, M(expected.Args, expected.Err), M(ParseShell(input)))
+		})
+	}
+}
 
-	cases := map[string][]string{
-		"":                  []string{},
-		"one":               []string{"one"},
-		"one two":           []string{"one", "two"},
-		"\"one\"":           []string{"one"},
-		"'one'":             []string{"one"},
-		"one two three":     []string{"one", "two", "three"},
-		"one \"two\" three": []string{"one", "two", "three"},
-		"one \"two three\"": []string{"one", "two three"},
-		"one 'two three'":   []string{"one", "two three"},
-		"\"one two three\"": []string{"one two three"},
-		"'one two three'":   []string{"one two three"},
-		"‘one two three’":   []string{"one two three"},
-		"“one two three”":   []string{"one two three"},
-		"one 'two three":    []string{"one", "'two", "three"},
-		"one \"two three":   []string{"one", "\"two", "three"},
+func TestParseShell_UserInputError(t *testing.T) {
+	cases := map[string]ParseShellOutput{
```

Comment:
When writing unit tests, consider the following:
* What behavior(s) am I testing? 
* Do I need multiple test cases? If so, which things change between tests?
* How do you write this test in the simplest way possible?

For this scenario:
* The behavior we are testing is that `ParseShell()` should turn a non-nil error on bad inputs. That means the bottom of the test should look like this:
```go
if _, err := ParseShell(...); err == nil {
    t.Fatal("Error was nil!")
}
```
* We are testing bad inputs, each of which is expected to result in the same behavior (returning a non-nil error), so it makes sense to have multiple test cases. 
* I tend to find that using maps and slices allow you to write test cases in the simplest way possible. In our case, our inputs are simple objects: strings. We don't have any other things that change between test cases, so we should use a `[]string` for our test inputs. For more complex objects, use [anonymous structs](https://dave.cheney.net/2013/06/09/writing-table-driven-tests-in-go).

Putting that together, our test should look something like this:
```go
func TestParseShellErrors(t *testing.T){
    inputs := []string{...}

    for _, input := range inputs {
        t.Run(input, func(t *testing.T){
            if _, err := ParseShell(input); err == nil {
                t.Fatal("Error was nil!")
            }
        })
    }
}
```